lib/ansible/modules/cloud/amazon/ec2_snapshot.py

```diff
@@ -239,16 +260,40 @@ def create_snapshot(module, ec2, state=None, description=None, wait=None,
             if not _create_with_wait(snapshot, wait_timeout):
                 module.fail_json(msg='Timed out while creating snapshot.')
         if snapshot_tags:
-            for k, v in snapshot_tags.items():
+            tags_to_add, tags_to_remove = existing_to_desired(snapshot.tags.items(), snapshot_tags.items())
+            for (k, v) in tags_to_add:
                 snapshot.add_tag(k, v)
+                changed = True
+            for (k, v) in tags_to_remove:
+                snapshot.remove_tag(k, value=v)
+                changed = True
+
+        permissions = ec2.get_snapshot_attribute(snapshot.id)
+
+        users_to_add, users_to_remove = existing_to_desired(permissions.attrs.get('user_ids', []),
```

Comment:
If you are removing the last user, users_to_add == ['None'], which is truthy so it fails on line 280. Same problem with remove the group "all".